sdk/typespec/typespec_client_core/src/error/http_error.rs

```diff
@@ -24,7 +24,7 @@ impl HttpError {
     /// Create an error from an HTTP response.
     ///
     /// This does not check whether the response was successful and should only be used with unsuccessful responses.
-    pub async fn new(response: RawResponse) -> Self {
+    pub async fn new(response: RawResponse, error_header_name: Option<HeaderName>) -> Self {
```

Comment:
1. I don't like the long parameter name. Since "error" is redundant, just "header_name" is probably fine.
2. That said, I don't think a single one is sufficient. IIRC, some services - like Storage - have a list they go through. Should this maybe be a `&[&'static str]`? Would `'static` be sufficient here since we know the list we'll pass ahead of time?

Reply:
I'd rename all the `error_header_name`. "error" just seems unnecessarily long.

Reply:
FWIW, the current typespec_client_core logic has no extensibility support for additional headers.

I'm not opposed to the idea of extending this to multiple headers, but as things currently stand, there is no need for more than the single header.

Also, making this service extensible is a non-trivial modification to the current design - the current design intentionally isolates the HTTP headers associated with the operation to the `azure_core` layer and intentionally removes that from service interactions.

Reply:
Followup: I'll add this extensibility as a part of the typespec_client_core::http::Error cleanup.